rust/hyperlane-base/src/settings/loader/arguments.rs

```diff
@@ -77,12 +86,15 @@ impl Source for CommandLineArguments {
                 continue;
             }
 
-            let mut key = key.to_lowercase();
-
-            // If separator is given replace with `.`
-            if !separator.is_empty() && separator != "." {
-                key = key.replace(separator, ".");
-            }
+            let key = if let Some(case) = self.casing {
+                // if case is given, replace case of each key component
+                key.split(separator).map(|s| s.to_case(case)).join(".")
+            } else if !separator.is_empty() && separator != "." {
+                // If separator is given replace with `.`
```

Comment:
I think this comment still applies to the entire `let key` block since the first conditional branch also does `.join(".")`